platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDb.java

```diff
@@ -205,6 +212,22 @@ private MerkleDb(final Path storageDir) {
         storeMetadata();
     }
 
+    /**
+     * Iterates over the list of table metadata starting from index from {@link #nextTableId} until
+     * it finds an ID that isn't used by a table.
+     *
+     * @return The next available table ID
+     */
+    private int getNextTableId() {
+        for (int tablesCount = 0; tablesCount < MAX_TABLES; tablesCount++) {
+            final int id = Math.abs(nextTableId.getAndIncrement() % MAX_TABLES);
```

Comment:
Why are we taking the absolute value here? It's not obvious to me how `nextTableId` can be negative.

Reply:
`nextTableId` may overflow. I don't think it will ever be the case, even with all data source copies to use different table IDs, but the chances are not zero.